test/unittests/test_zx_functionality.cpp

```diff
@@ -120,7 +120,6 @@ TEST_F(ZXDiagramTest, complex_circuit) {
     zx::fullReduce(diag);
     EXPECT_EQ(diag.getNVertices(), 6);
     EXPECT_EQ(diag.getNEdges(), 3);
-    EXPECT_TRUE(diag.isIdentity());
```

Comment:
shouldn't this check be replaced with something appropriate?

Reply:
Definitely. I have adjusted this accordingly